test/core.jl

```diff
@@ -4822,6 +4822,17 @@ end
 @test f14893() == 14893
 @test M14893.f14893() == 14893
 
+# issue #18725
+@test_nowarn begin
```

Comment:
(My workaround now evals the expression in `Main`, so it correctly gets the case that is not supposed to have a warning.)